custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry;
+
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.HashSet;
+import java.util.Set;
+
+public class JdbcSpanRenamingProcessor implements SpanProcessor {
+  private static final Set<String> SQL_SYSTEMS;
+
+  static {
+    Set<String> sqlSystems = new HashSet<>();
+    sqlSystems.add("db2");
+    sqlSystems.add("derby");
+    sqlSystems.add("h2");
+    sqlSystems.add("hsqldb");
+    sqlSystems.add("mariadb");
+    sqlSystems.add("mssql");
+    sqlSystems.add("mysql");
+    sqlSystems.add("oracle");
+    sqlSystems.add("other_sql");
+    sqlSystems.add("postgresql");
+    SQL_SYSTEMS = sqlSystems;
+  }
+
+  @Override
+  public void onStart(ReadWriteSpan span) {
+    AttributeValue dbSystem = span.toSpanData().getAttributes().get("db.system");
+    if (isJdbcSpan(dbSystem)) {
+      String query = span.getName().trim().toUpperCase();
+      span.updateName(getStatementType(query));
+    }
+  }
+
+  private boolean isJdbcSpan(AttributeValue dbSystem) {
+    return dbSystem != null
+        && dbSystem.getType() == AttributeValue.Type.STRING
+        && SQL_SYSTEMS.contains(dbSystem.getStringValue());
+  }
+
+  private String getStatementType(String query) {
```

Comment:
I think if you change this to `SELECT ` etc, then you can ditch `isJdbcSpan`

Reply:
Hmm, you mean detecting SQL spans just by the prefix? Hmm, I think it's a bit safer this way - this processor won't rename any random span (not necessarily DB related) whose name begins with `select ` (e.g. span manually added by the instrumented app).

Reply:
Ok, makes sense. I just did not like this hard-copied list of db systems :(